lib/widgets/random_book.dart

```diff
@@ -0,0 +1,116 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart' show rootBundle;
+import 'dart:convert';
+import '../models/book.dart';
+
+class RandomBookWidget extends StatefulWidget {
+  const RandomBookWidget({Key? key}) : super(key: key);
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _RandomBookWidgetState createState() => _RandomBookWidgetState();
+}
+
+class _RandomBookWidgetState extends State<RandomBookWidget> {
+  Book? book;
+
+  @override
+  void initState() {
+    super.initState();
+    loadBooks();
+  }
+
+  void loadBooks() async {
+    final jsonFile = await rootBundle.loadString('assets/books.json');
+    final parsedJson = json.decode(jsonFile);
```

Comment:
This operation is done in the _DiscoverPageState class as well. So you want to use the existing `List<Book> _books` in the  _DiscoverPageState class. To do it, you need to take the list with the constructor of the RandomBookWidget class.